File: Codes in C++/Dynamic Programming/Frog Jump 1-2 steps energy level.cpp
#include <bits/stdc++.h> 
int frogJump(int n, vector<int> &heights)
{
    int prev=0;
    int prev2=0;
    for(int i=1;i<n;i++)
    {
        int fs=prev+abs(heights[i]-heights[i-1]);
        int ss=INT_MAX;
        if(i>1)
            ss=prev2+abs(heights[i]-heights[i-2]);
        int cur=min(fs,ss);
        prev2=prev;
        prev=cur;
    }
    return prev;
}

File: Codes in Python3/Dynamic Programming/Frog Jump 1-2 steps energy level.py
from os import *
from sys import *
from collections import *
from math import *

from typing import *

  
def frogJump(n: int, heights: List[int]) -> int:

    prev=0
    prev2=0
    for i in range(1,n):
        fs=prev+abs(heights[i]-heights[i-1])
        ss=inf
        if i>1:
            ss=prev2+abs(heights[i]-heights[i-2])
        cur=min(fs,ss)
        prev2=prev
        prev=cur
    return prev
    pass
